src/components/layouts/MainLayout.jsx

```diff
@@ -31,6 +34,8 @@ export const MainLayout = () => {
 
   const location = useLocation()
   const [selectedRoute, setSelectedRoute] = useState(location.pathname || ROUTES.CALENDAR)
+  const searchParams = new URLSearchParams(location.search)
```

Comment:
for working with query params there is useSearchParams hook. Please check how it works